### DISCUSSION TITLE
how to set automation trigger when my front-end and back-end repo have codecommit?

### DISCUSSION THREAD

**Discussion by Bagel-tvbs** [codeceptjs/CodeceptJS#3685] [Q&A] [answered]
I have three repo 1: front-end code 2: back-end code 3: codeceptjs and locust (for automation testing)
I want to ask how can I do to automatic trigger a automation testing when front-end and backend code have a commit?

**Comment by EgorBodnar** [comment, marked as answer]
### This question is not related to the Codecept js framework, but here's where you need to move:

- You need to build a ci/cd pipeline of projects so that you can listen to the FE/BE repositories and build the FE/BE on push/PR creation, clone out your auto-test framework and run tests.

- Or you can do this on `pre-commit` and run tests locally (see [husky](https://typicode.github.io/husky/)), but the process of developers committing to the project slows down so much that developers will just start skipping this step.

### If you do decide to do it on the CI/CD side, here's what you need to learn depending on the CI/CD system itself on your project:

1. **Jenkins Pipeline**:  You can set up Jenkins to automatically run the pipeline when you push a commit by using the GitHub plugin and configuring a webhook in your GitHub repository settings. 
   
   - [Jenkins Pipeline Tutorial](https://www.jenkins.io/doc/book/pipeline/)
   - [Jenkins GitHub Plugin](https://plugins.jenkins.io/github/)

2. **GitHub Actions**: You can set up GitHub Actions to automatically run tests or builds when you push a commit or create a PR by using workflows. Define these workflows in YAML files in the `.github/workflows` directory in your repository.

   - [GitHub Actions Documentation](https://docs.github.com/en/actions)
   
3. **GitLab CI/CD**: Set GitLab to automatically run the pipeline when you push a commit or create a merge request. 

   - [GitLab CI/CD Documentation](https://docs.gitlab.com/ee/ci/)

4. **Bamboo**:  Use repository triggers to run builds after code gets committed to a repository. 

   - [Bamboo Documentation](https://confluence.atlassian.com/bamboo/bamboo-documentation-289276898.html)

5. **CircleCI**: Set up CircleCI to automatically build when you push a commit.

   - [CircleCI Documentation](https://circleci.com/docs/2.0/)

6. **Travis CI**: Set up Travis CI to automatically run tests when you push a commit.

   - [Travis CI Documentation](https://docs.travis-ci.com/)

**Reply by kobenguyent** [reply]
you could try this https://github.com/kobenguyent/flowmatic which is a very basic approach to create the ci/cd configurations
